fecalc.js

```diff
@@ -0,0 +1,292 @@
+    if ( ( (characters[0].flags.includes("n") && characters[1].flags.includes("n")) || (!characters[0].flags.includes("n") && !characters[1].flags.includes("n")) ) || (characters[0].flags.includes("n") && characters[1].flags.includes("o")) ) {
+        results += color[1] + "retaliates for " + calc[2]  + "!\n";
+        results += color[0] + "is " + ( (calc[3] <= 0) ? "killed!\n" : "left with " + calc[3] + "!\n" );
+    } else {
+        results += color[1] + "cannot attack back!\n";
+    }
+    // Speed Check for Extra Attack / Counter 
+    if ( (characters[0].spd - characters[1].spd) >= 5 ) {
+        results += color [0] + "attacks again for " + calc[0] + "!\n";
+        results += color[1] + "is " + ( ((calc[1] - calc[0]) <= 0) ? "killed!\n" : "left with " + ( calc[1] - calc[0]) + "!\n" );
+    } else if ( (characters[0].spd - characters[1].spd) >= 5 ) {
+        if ( ( (characters[0].flags.includes("n") && characters[1].flags.includes("n")) || !(characters[0].flags.includes("n") && characters[1].flags.includes("n")) ) || (characters[0].flags.includes("n") && characters[1].flags.includes("o")) ) {
+            results += color[1] + "retaliates again for " + calc[2]  + "!\n"; 
+            results += color[0] + "is " + ( ((calc[3] - calc[2]) <= 0) ? "killed!\n" : "left with " + (calc[3] - calc[2]) + "!\n" );
+        }     
+    }
+    if (typeof callback === "function" ) {
+        callback ( results );
+    }
+}
+
+function errorMissingColor() {
+    return "Missing color designation";
+}
+
+// Tests are not used
+//
+// feParseTest is a Custom Unit Test for function feParse()
+function feParseTest () {
+    var results = { total: 0, bad: 0};
+    
+    function test( ran, expected ) {
+       results.total++;
+       var running = feParse(ran);
+       for ( var i = 0; i < running.length; i++ ){
+           if ( running[i] != expected[i])
+           {
+              results.bad++; 
+              console.log( "Expected " + expected + " but returned " + running );
+              break; 
+           }
+       }
+    }
+
+    test( "3849261735gnvs3035263024b", ["3849261735gn", "3035263024b",] );
+    test( "3849261735gnv3035263024b", ["3849261735gn", "3035263024b",] );
+    test( "gn3849261735vb3035263024", ["gn3849261735", "b3035263024",] );
+    test( "g3849261735nvsb3035263024", ["g3849261735n", "b3035263024",] );
+    test( "g3849261735nv3035263024b", ["g3849261735n", "3035263024b",] );
+    test( "g 38 49 26 17 35 n v 30 35 26 30 24 b", ["g3849261735n", "3035263024b",] );
+    test( "g 38 49 26 17 35 n vs 30 35 26 30 24 b", ["g3849261735n", "3035263024b",] );
+    test( "g n 38 49 26 17 35  vs b 30 35 26 30 24", ["gn3849261735", "b3035263024",] );
+    test( "3849261735v3035263024", ["3849261735", "3035263024",] );
+    console.log( results.total - results.bad + " passed. " + results.bad + " failed out of " + results.total);
+}
+
+// feCharTest is a Custom Unit Test for function feCreateChar()
+function feCharTest () {
```

Comment:
Are you calling these test functions anywhere? If so, remove the calls triggering these functions. Good job making unit tests.